File: frontend/lib/models/financeiro.dart
class Financeiro {
  final int id;
  final int processoId;
  final int clienteId;
  final String tipo;
  final String? descricao;
  final double valor;
  final String status;
  final String? dataVencimento;
  final String? dataPagamento;
  final DateTime createdAt;
  final DateTime updatedAt;

  Financeiro({
    required this.id,
    required this.processoId,
    required this.clienteId,
    required this.tipo,
    this.descricao,
    required this.valor,
    required this.status,
    this.dataVencimento,
    this.dataPagamento,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Financeiro.fromJson(Map<String, dynamic> json) {
    return Financeiro(
      id: json['id'],
      processoId: json['processo_id'],
      clienteId: json['cliente_id'],
      tipo: json['tipo'],
      descricao: json['descricao'],
      valor: (json['valor'] as num).toDouble(),
      status: json['status'],
      dataVencimento: json['data_vencimento'],
      dataPagamento: json['data_pagamento'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'processo_id': processoId,
      'cliente_id': clienteId,
      'tipo': tipo,
      'descricao': descricao,
      'valor': valor,
      'status': status,
      'data_vencimento': dataVencimento,
    };
  }
}

class FinanceiroResumo {
  final double pendente;
  final double pago;
  final double total;

  FinanceiroResumo({
    required this.pendente,
    required this.pago,
    required this.total,
  });

  factory FinanceiroResumo.fromJson(Map<String, dynamic> json) {
    return FinanceiroResumo(
      pendente: (json['pendente'] as num).toDouble(),
      pago: (json['pago'] as num).toDouble(),
      total: (json['total'] as num).toDouble(),
    );
  }
}

File: backend/app/services/ferramentas/financeiro.py
from sqlalchemy.orm import Session
from app.models import Financeiro


SCHEMA_RESUMO_FINANCEIRO = {
    "name": "resumo_financeiro",
    "description": "Resume a posicao financeira de um processo especifico. Mostra totais pendentes, pagos e lancamentos detalhados.",
    "input_schema": {
        "type": "object",
        "properties": {
            "processo_id": {
                "type": "integer",
                "description": "ID do processo",
            },
        },
        "required": ["processo_id"],
    },
}


def executar_resumo_financeiro(input_data: dict, db: Session) -> str:
    processo_id = input_data["processo_id"]

    lancamentos = (
        db.query(Financeiro)
        .filter(Financeiro.processo_id == processo_id)
        .order_by(Financeiro.data_vencimento)
        .all()
    )

    if not lancamentos:
        return f"Nenhum lancamento financeiro para o processo ID {processo_id}."

    total_pendente = sum(float(f.valor) for f in lancamentos if f.status == "pendente")
    total_pago = sum(float(f.valor) for f in lancamentos if f.status == "pago")

    linhas = [
        f"FINANCEIRO DO PROCESSO #{processo_id}:",
        f"  Total pendente: R$ {total_pendente:,.2f}",
        f"  Total pago: R$ {total_pago:,.2f}",
        f"  Total geral: R$ {total_pendente + total_pago:,.2f}",
        "",
        "LANCAMENTOS:",
    ]
    for f in lancamentos:
        venc = f.data_vencimento.strftime("%d/%m/%Y") if f.data_vencimento else "S/D"
        linhas.append(f"  [{f.status.upper()}] {f.tipo} — R$ {float(f.valor):,.2f} — Venc: {venc} — {f.descricao or ''}")

    return "\n".join(linhas)
